<div style="height: 200px; background-image: url('https://legalserviceindia.com/legal/uploads/offencesagainstchildrenunderipc_2824045185.jpg'); background-size: cover; background-position: center;"></div>




# Exploratory Data Analysis

- This notebook presents an analysis of crimes against children in India for the year 2014. The dataset, sourced from the Government of India's official crime statistics, provides detailed information on various types of crimes reported across different states and union territories. The goal of this analysis is to uncover patterns, trends, and regional variations in crimes against children, with a specific focus on relevant crime categories.

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
crime_csv = 'cleaned_data.csv'
df = pd.read_csv('cleaned_data.csv')

In [7]:
df.head(5)

States/UTs                                         Crime Head  2014
0  Andhra Pradesh               1 - Murder (Section 302 and 303 IPC)    45
1  Andhra Pradesh                  2 - Infanticide (Section 315 IPC)     2
2  Andhra Pradesh                                           3 - Rape   477
3  Andhra Pradesh  4 - Assault on women with intent to outrage he...   274
4  Andhra Pradesh  5 - Insult to the Modesty of Women (Girls Chil...    75

### 1) Total number of crimes against children happend in the year 2014 in India

In [9]:
total_crimes_2014 = df['2014'].sum()
print ('Total number of crimes against children in 2014 is ',total_crimes_2014)

Total number of crimes against children in 2014 is  89316


- Total number of crimes against children in 2014 is  89316

### 2) Total number of crimes against children state and union wise

In [10]:
crimes_by_state = df.groupby('States/UTs')['2014'].sum().reset_index()
print(crimes_by_state)


           States/UTs   2014
0         A&N Islands     50
1      Andhra Pradesh   2059
2   Arunachal Pradesh    134
3               Assam   1385
4               Bihar   2255
5          Chandigarh    207
6        Chhattisgarh   4353
7          D&N Haveli     11
8         Daman & Diu      7
9            Delhi UT   9345
10                Goa    330
11            Gujarat   3219
12            Haryana   2534
13   Himachal Pradesh    463
14    Jammu & Kashmir    211
15          Jharkhand    422
16          Karnataka   3416
17             Kerala   2391
18        Lakshadweep      1
19     Madhya Pradesh  15055
20        Maharashtra   8108
21            Manipur    137
22          Meghalaya    213
23            Mizoram    178
24           Nagaland     25
25             Odisha   2196
26         Puducherry     38
27             Punjab   1752
28          Rajasthan   3856
29             Sikkim     93
30         Tamil Nadu   2354
31          Telangana   1928
32            Tripura    369
33      Uttar 

### 3) The state/UT with highest number of crimes against children

In [22]:
crimes_by_state_sorted = crimes_by_state.sort_values(by='2014', ascending=False).reset_index(drop=True)
crimes_by_state_sorted

States/UTs   2014
0      Madhya Pradesh  15055
1       Uttar Pradesh  14824
2            Delhi UT   9345
3         Maharashtra   8108
4         West Bengal   4909
5        Chhattisgarh   4353
6           Rajasthan   3856
7           Karnataka   3416
8             Gujarat   3219
9             Haryana   2534
10             Kerala   2391
11         Tamil Nadu   2354
12              Bihar   2255
13             Odisha   2196
14     Andhra Pradesh   2059
15          Telangana   1928
16             Punjab   1752
17              Assam   1385
18        Uttarakhand    488
19   Himachal Pradesh    463
20          Jharkhand    422
21            Tripura    369
22                Goa    330
23          Meghalaya    213
24    Jammu & Kashmir    211
25         Chandigarh    207
26            Mizoram    178
27            Manipur    137
28  Arunachal Pradesh    134
29             Sikkim     93
30        A&N Islands     50
31         Puducherry     38
32           Nagaland     25
33         D&N Haveli     11
34        Daman & Diu      7
35        Lakshadweep      1

- Madhya pradesh recorded highest number of crimes and lakshadweep reported very few. Since the population is different in each state and unuin territory, we can't make very well assumptions

### 3. Sum of Crimes by Crime Head

In [24]:
crimes_by_crime_head = df.groupby('Crime Head')['2014'].sum().reset_index()
crimes_by_crime_head_sorted = crimes_by_crime_head.sort_values(by='2014', ascending=False).reset_index(drop=True)
crimes_by_crime_head_sorted

Crime Head   2014
0   6 - Kidnapping & Abduction_Total (Section 363,...  37854
1                                            3 - Rape  13766
2   4 - Assault on women with intent to outrage he...  11335
3   19 - Protection of Children from Sexual Offenc...   8904
4        22 - Other crimes committed against children   8484
5   10 - Procuration of minor girls (Section 366-A...   2020
6                1 - Murder (Section 302 and 303 IPC)   1817
7   18 - Juvenile Justice (Care and Protection of ...   1315
8      9 - Exposure and Abandonment (Section 317 IPC)    983
9           20 - Attempt to commit Murder u/s 307 IPC    840
10                21 - Unnatural Offences u/s 377 IPC    765
11  5 - Insult to the Modesty of Women (Girls Chil...    444
12       14 - Prohibition of Child Marriage Act, 2006    280
13  16 - Child Labour (Prohibition & Regulation) A...    147
14                  2 - Infanticide (Section 315 IPC)    121
15        17 - Immoral Traffic (Prevention) Act, 1956     86
16  13 - Selling of minors for prostitution (Secti...     82
17  8 - Abetment of Suicide of child (Section 305 ...     56
18  12 - Buying of minors for prostitution (Sectio...     14
19  11 - Importation of Girls from Foreign Country...      2
20     15 - Transplantation of Human Organs Act, 1994      1

- kidnapping and abduction cases is the most reported cases and the number of rape cases is alarming

### 4. Total number of crimes recorded in Assam

In [29]:
specific_state = 'Assam'  # Replace with the state/UT of interest
crime_stats_state = df[df['States/UTs'] == specific_state].set_index('Crime Head')['2014']

total_crimes_specific_state = crime_stats_state.sum()

print(f"Total number of crimes in {specific_state} in 2014:", total_crimes_specific_state)


Total number of crimes in Assam in 2014: 1385


### 5. Total number of murders by state

In [34]:
murder_head = '1 - Murder (Section 302 and 303 IPC)'
murder_df = df[df['Crime Head'] == murder_head]

# Group by State/UT and sum the number of crimes
sum_murder_by_state = murder_df.groupby('States/UTs')['2014'].sum()

# Sort by the number of crimes in ascending order
sorted_murder_states = sum_murder_by_state.sort_values(ascending=False)

sorted_murder_states

States/UTs
Uttar Pradesh        543
Maharashtra          195
Madhya Pradesh       131
Tamil Nadu           103
Karnataka             87
Gujarat               82
Rajasthan             73
Telangana             70
West Bengal           64
Bihar                 61
Haryana               56
Chhattisgarh          50
Delhi UT              48
Andhra Pradesh        45
Kerala                44
Punjab                42
Odisha                23
Assam                 22
Tripura               21
Uttarakhand           11
Manipur                9
Jharkhand              8
Himachal Pradesh       7
Meghalaya              7
Arunachal Pradesh      5
Goa                    3
Jammu & Kashmir        3
Sikkim                 1
A&N Islands            1
Nagaland               1
D&N Haveli             1
Puducherry             0
Mizoram                0
Daman & Diu            0
Chandigarh             0
Lakshadweep            0
Name: 2014, dtype: int64

- Uther pradesh recorded more child murders with a great gap with the second state

### 6. Total number of Rapes by state

In [35]:
rape_head = '3 - Rape'
rape_df = df[df['Crime Head'] == rape_head]

# Group by State/UT and sum the number of crimes
sum_rape_by_state = rape_df.groupby('States/UTs')['2014'].sum()

# Sort by the number of crimes in ascending order
sorted_rape_states = sum_rape_by_state.sort_values(ascending=False)

sorted_rape_states

States/UTs
Madhya Pradesh       2352
Maharashtra          1714
Uttar Pradesh        1538
Delhi UT             1004
Rajasthan             825
Chhattisgarh          806
Kerala                754
Odisha                753
Karnataka             694
Telangana             583
Andhra Pradesh        477
Punjab                476
Haryana               416
Gujarat               291
Himachal Pradesh      135
Assam                 125
Tripura               122
Bihar                 103
Mizoram                91
Uttarakhand            84
Jharkhand              78
Meghalaya              61
Goa                    57
Arunachal Pradesh      43
Sikkim                 40
Manipur                38
Jammu & Kashmir        37
Chandigarh             32
A&N Islands            20
Nagaland               11
Puducherry              3
D&N Haveli              2
Daman & Diu             1
Lakshadweep             0
Tamil Nadu              0
West Bengal             0
Name: 2014, dtype: int64

- Madhya pradesh recorded most nuber of child rape cases in the year 2014

### 7. Is child marriage is a thing in 2014 india?

In [36]:
marriage_head = '14 - Prohibition of Child Marriage Act, 2006'
marriage_df = df[df['Crime Head'] == marriage_head]

# Group by State/UT and sum the number of crimes
sum_marriage_state = marriage_df.groupby('States/UTs')['2014'].sum()

# Sort by the number of crimes in ascending order
sorted_marriage_states = sum_marriage_state.sort_values(ascending=False)

sorted_marriage_states

States/UTs
Tamil Nadu           47
Karnataka            44
West Bengal          37
Kerala               19
Andhra Pradesh       16
Gujarat              16
Madhya Pradesh       15
Haryana              15
Maharashtra          14
Telangana            13
Uttar Pradesh        10
Bihar                10
Rajasthan             5
Himachal Pradesh      4
Assam                 4
Puducherry            3
Delhi UT              2
Chhattisgarh          2
Punjab                2
Tripura               1
Jharkhand             1
Odisha                0
Sikkim                0
Uttarakhand           0
A&N Islands           0
Nagaland              0
Mizoram               0
Meghalaya             0
Manipur               0
Jammu & Kashmir       0
Goa                   0
Daman & Diu           0
D&N Haveli            0
Chandigarh            0
Arunachal Pradesh     0
Lakshadweep           0
Name: 2014, dtype: int64

- Child marriage still prevails in india (2014)
- Tamil nadu recorded hghest number of child marriage
- 15 states are not recorded any child marriages

### 8.Total number of child abduction by state

In [40]:
abduction_head = '6 - Kidnapping & Abduction_Total (Section 363,364,364A, 366-369 IPC)'
abduction_df = df[df['Crime Head'] == abduction_head]

# Group by State/UT and sum the number of crimes
sum_abduction_state = abduction_df.groupby('States/UTs')['2014'].sum()

# Sort by the number of crimes in ascending order
sorted_abduction_states = sum_abduction_state.sort_values(ascending=False)

sorted_abduction_states

States/UTs
Delhi UT             6452
Madhya Pradesh       6339
Uttar Pradesh        5875
Maharashtra          2616
West Bengal          2351
Gujarat              2101
Chhattisgarh         1844
Rajasthan            1814
Bihar                1585
Karnataka            1177
Punjab                871
Haryana               816
Odisha                800
Telangana             634
Andhra Pradesh        600
Tamil Nadu            390
Uttarakhand           275
Himachal Pradesh      210
Assam                 152
Jammu & Kashmir       139
Goa                   134
Kerala                130
Chandigarh            130
Jharkhand              94
Tripura                88
Manipur                69
Arunachal Pradesh      66
Meghalaya              43
Sikkim                 17
A&N Islands            12
Puducherry              9
D&N Haveli              8
Nagaland                7
Daman & Diu             4
Mizoram                 2
Lakshadweep             0
Name: 2014, dtype: int64

- Delhi UT,madhya pradesh and uttar pradesh recorded high number of child abduction cases in india in 2014
- mizoram is the state that recorded very least chid abduction cases.
